sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -368,190 +560,337 @@ public virtual PartitionProperties CreatePartitionPropertiesFromResponse(AmqpMes
         ///
         private static EventData BuildEventFromAmqpMessage(AmqpMessage source)
         {
-            var body = (source.BodyType.HasFlag(SectionFlag.Data))
-                ? ReadAmqpDataBody(source.DataBody)
-                : new BinaryData(ReadOnlyMemory<byte>.Empty);
+            var message = source switch
+            {
+                _ when TryGetDataBody(source, out var dataBody) => new AmqpAnnotatedMessage(dataBody),
+                _ when TryGetSequenceBody(source, out var sequenceBody) => new AmqpAnnotatedMessage(sequenceBody),
+                _ when TryGetValueBody(source, out var valueBody) => new AmqpAnnotatedMessage(valueBody),
+                _ => new AmqpAnnotatedMessage(AmqpMessageBody.FromData(MessageBody.FromReadOnlyMemorySegment(ReadOnlyMemory<byte>.Empty)))
```

Comment:
For cases where the body type cannot be determined, assume an empty data body; this is necessary to preserve consistency with existing behavior.